src/foundation.cs

```diff
@@ -8194,16 +8194,16 @@ interface NSInputStream {
 
 	}
 
-	delegate bool EnumerateLinguisticTagsEnumerator (NSString tag, NSRange tokenRange, NSRange sentenceRange, ref bool stop);
+	delegate bool NSEnumerateLinguisticTagsEnumerator (NSLinguisticTagScheme tag, NSRange tokenRange, NSRange sentenceRange, ref bool stop);
```

Comment:
I don't think this will work - we don't have the logic to convert from a smart enum (to `NSString`) for the delegate
Can you write a test ?